### DISCUSSION TITLE
Video Depth Anything - model support?

### DISCUSSION THREAD

**Discussion by n1Stark** [nagadomi/nunif#307] [Q&A] [answered]
hi,
first of all - huge thanks for this wonderful software. your efforts are greatly appreciated.

my question is if there is any list of models supported by the software. im not an avid CLI user. however using the gui version i cannot find the new "Video Depth Anything" model in the list. i've understood some models are disabled and others are not, not sure how to determine whats ok and not..

is there any way for me to enable the model or to check out a supported model list?

**Comment by nagadomi** [comment, marked as answer]
VideoDepthAnything is not yet supported.
There are two problems with that.

1. Too many VRAM requirements.
2. Does not support stream input (sequential frame input online processing)

1 has been resolved, but with 8G VRAM large model only works up to 392 resolution.
2 is not yet supported. Without this, it can't handle truly super long videos, like 2 hours.

Unhappy to say that my keyboard broke yesterday and I am currently waiting for a new one to be delivered.

**Reply by n1Stark** [reply]
i see, thanks alot for the answer.
sad about your keyboard hope you'll get a new keyboard shortly✌️

**Comment by nagadomi** [comment]
current status

 1. is merged https://github.com/DepthAnything/Video-Depth-Anything/pull/24
2. is implemented https://github.com/nagadomi/Video-Depth-Anything_iw3/commit/becaea9e81c35be21ac85742e48062d6caa4a1b6

scene detection is implemented https://github.com/nagadomi/nunif/issues/311#issuecomment-2700849770

Already have all the parts needed.

**Reply by laiyangli001** [reply]
Thank you for hard work.
Does this mean can we use this "Video depth anthing" model in iw3 desktop ,when watching movie on computer,it will be better than other mode?

**Reply by nagadomi** [reply]
video depth anything requires 32 frame batches for temporal attention.
That means at 30 fps it would cause more than 1 second latency.
So it cannot be used for real-time (low-latency) applications.

**Reply by nagadomi** [reply]
It can be used in iw.3desktop if the following conditions are met
- A few seconds of latency is acceptable
- Streaming audio synchronized with video

However, currently iw3.desktop does not support streaming audio.

**Reply by QuickscopingFTW** [reply]
> current status
> 
> 1. is merged [Split batch_size(=32) into micro_batch_size(=4) in DPTHeadTemporal DepthAnything/Video-Depth-Anything#24](https://github.com/DepthAnything/Video-Depth-Anything/pull/24)
> 2. is implemented [nagadomi/Video-Depth-Anything_iw3@becaea9](https://github.com/nagadomi/Video-Depth-Anything_iw3/commit/becaea9e81c35be21ac85742e48062d6caa4a1b6)
> 
> scene detection is implemented [#311 (comment)](https://github.com/nagadomi/nunif/issues/311#issuecomment-2700849770)
> 
> Already have all the parts needed.

is there any rough eta on when this will be added?

**Comment by C-Three-P-O** [comment]
Can Video-Depth_anything be used in IW3?
I noticed Video-Depth-Anything_iw3, downloaded the pre-trained models in the checkpoints directory like i did with the Distill_Any models.
They don't show up in the programs list. So is it supported? I use the Windows package; do i have to do something else? Or does Video-Depth-Anything_iw3 only works under Linux?

**Comment by laiyangli001** [comment]
[nunif_modify.zip](https://github.com/user-attachments/files/20202115/nunif_modify.zip)

![QQ_1747205007983](https://github.com/user-attachments/assets/4adfdf88-38e6-4b2d-802d-0dbda26eb47b)
![QQ_1747205176955](https://github.com/user-attachments/assets/8a58e684-e74f-4cb4-b53c-21b20612625d)
![QQ_1747205248752](https://github.com/user-attachments/assets/ee3b66b6-44f9-4c0e-b8b2-08f39af849ce)
![QQ_1747205449446](https://github.com/user-attachments/assets/6b1ab215-2b65-493b-b4cc-402ed6900192)
@n1Stark  @nagadomi I attepmed to add "Video depth anthing" model to iw3 desktop ,and seems running successfully,how do you think about?

**Reply by nagadomi** [reply]
@laiyangli001 
You don't seem to understand anything.
You need to understand the difference between DepthAnythingV2 and VideoDepthAnything first. You're already mistaken at that point. Once you understand the difference, you need to implement that difference in the code.

**Reply by laiyangli001** [reply]
Yes ,you are right,so I am not sure about this.

**Reply by nagadomi** [reply]
As I mentioned above, using VideoDepthAnything with iw3.desktop isn't very practical. You've probably heard the same opinion from the author of DeepDesktop as well.
I might add it to iw3, but I have no plans to add it to iw3.desktop.

**Comment by francdn** [comment]
Hi @nagadomi . I was going to check VideoDepthAnything support from the dev brach and I noticed something that could be an error.

file: video_depth_anything_model.py

line 19: 
MODEL_FILES = {
    "VDA_S": path.join(HUB_MODEL_DIR, "checkpoints", "video_depth_anything_vits.pth"),
    "VDA_L": path.join(HUB_MODEL_DIR, "checkpoints", "video_depth_anything_vitl.pth"),
    "VDA_Metric": path.join(HUB_MODEL_DIR, "checkpoints", "video_depth_anything_vitl.pth"),
}

Both VDA_L and VDA_Metric refer to the same file name. Shouldn't VDA_Metric refer to "metric_video_depth_anything_vitl.pth"?

Also, one clarification. When you said that "video depth anything requires 32 frame batches for temporal attention.", does that mean that on the GUI " we have to set "Depth Batch Size" to 32?

**Reply by nagadomi** [reply]
@francdn 
You're right. the filename is wrong. But the correct file is actually loaded from 
https://github.com/nagadomi/Video-Depth-Anything_iw3/blob/7ceb644f90e2d89c3f4a753d388bd1290b798c4e/hubconf.py#L25-L26
so it works. The issue is with the existence check only. I fixed it.

>Also, one clarification. When you said that "video depth anything requires 32 frame batches for temporal attention.", does that mean that on the GUI " we have to set "Depth Batch Size" to 32?

`Batch Size=32` is automatically used internally. For depth estimation, the specified `Batch Size` is ignored.
However, the specified `Batch Size` is used during preprocessing and stereo generation.
Therefore, you don't need to explicitly set it to 32, but the specified value still has some effect.

**Reply by francdn** [reply]
@nagadomi 
>Therefore, you don't need to explicitly set it to 32, but the specified value still has some effect.

Does it have any effect on the resulting quality of the output (and if it does what would be the recommended value?) or only on  performance?

**Reply by nagadomi** [reply]
@francdn only on performance(processing time) and VRAM usage.

**Reply by nagadomi** [reply]
I’ve written some notes in https://github.com/nagadomi/nunif/tree/dev/iw3#vda-implementation-notes

**Reply by francdn** [reply]
> I’ve written some notes in https://github.com/nagadomi/nunif/tree/dev/iw3#vda-implementation-notes

Thanks a lot.

**Reply by diverswan9** [reply]
Will this eventually be integrated into the iw3 GUI? I added the models to checkpoints. Ran update .bat. Not showing in Depth Model section.

**Reply by francdn** [reply]
> Will this eventually be integrated into the iw3 GUI? I added the models to checkpoints. Ran update .bat. Not showing in Depth Model section.

It already appears on the GUI but you have to use the dev branch.

**Reply by diverswan9** [reply]
ok... thanks

**Reply by C-Three-P-O** [reply]
Also use the iw3-GUI, downloaded model to checkpoint, update.bat but not showing up. What do you mean with use the dev branch?

**Reply by diverswan9** [reply]
I ran **nunif-prompt.bat** and entered,

    git checkout -b dev origin/dev  

...then launched **iw3-gui.bat**. That seems to put it into dev-mode.
But get error "no module named einops" when trying to use any of the VDA models
So I ran **update-installer.bat**, then **update.bat**... it's now working.

**Reply by C-Three-P-O** [reply]
Yes, thanx allot

**Reply by C-Three-P-O** [reply]
![einops](https://github.com/user-attachments/assets/bf957f52-247f-4eb3-920e-4596634127ca)

how do i solve this?

**Reply by nagadomi** [reply]
After switching to the dev branch, please make sure to run update.bat.
https://github.com/nagadomi/nunif/blob/master/windows_package/docs/README.md#dev-branch

**Comment by nagadomi** [comment]
min-max normalization (depth range) will be improved later.
Currently, future frames are not referenced, so the depth range is updated with a delay (when using Flicker reduction).
It would be better to take future frames into account as well.

**Reply by diverswan9** [reply]
Scene Segmentation (--scene-segment)

How is this implemented in the GUI? I notice the 3D depth goes flat upon some scene changes.  Also the small model seems the most consistent 3d-wise.

**Reply by nagadomi** [reply]
Scene Segmentation will be applied if the option is checked.
However, even if it's enabled, if the depth range is not reset at scene changes, it's possible that the shot boundary detection is not working correctly due to accuracy issues.

To inspect the depth or scene boundaries, specify `Stereo Format = Debug Depth`.
This will output depth video, with a red line shown at the top of the frame to indicate a scene boundary (displayed for one frame only, VDA model only).
EDIT: This was incorrect. The red line is not shown when using the VDA model. The red line is displayed when using models other than VDA.

**Reply by diverswan9** [reply]
Oops! I missed the scene segmentaion checkbox.  Someone keeps adding new things. :)  

-thanks

**Reply by tufeixp** [reply]
Looks like the Scene Segmentation not work for me, when checked the option, the conersion will freeze in the loading VDA status.

**Reply by nagadomi** [reply]
It might be a network issue with downloading the model.
It downloads 
https://github.com/nagadomi/nunif/releases/download/0.0.0/transnetv2-pytorch-weights.pth
to
`iw3/pretrained_models/hub/checkpoints/transnetv2-pytorch-weights.pth`

**Reply by diverswan9** [reply]
Does Foreground Scale have the same effect for VDA as it does for Any V2? Or is it just ignored ?

**Comment by crossterra-agent** [comment]
Requested completion review: https://github.com/CrossTerra/monorepo/pull/2087#issuecomment-4104356080
